app/ui/SelectInput.tsx

```diff
@@ -24,11 +30,35 @@ export default function SelectInput({ className }: { className?: string }) {
   // state for current search string
   const [search, setSearch] = useState<string>("");
 
+  // state for currently selected options
+  const [val, setValue] = useState<string[]>(
+    searchParams.options
+      ? [searchParams.options].flat().join(",").split(",")
```

Comment:
Btw isn't join and split the same thing?

Reply:
As searchParams.options can be string or array of strings, we have to join potential array of strings with comma and then split it - commas are between the strings but also inside of them